### PR TITLE
os/bluestore: fix for FreeBSD iocb structure

### PR DESCRIPTION
**Pull request by wjwithagen**
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

### CODE REVIEW

File: src/os/bluestore/KernelDevice.cc [hunk 1]
```diff
@@ -521,7 +521,11 @@ void KernelDevice::_aio_thread()
 		devname.c_str(),
 		path.c_str(),
 		r,
-		aio[i]->iocb.aio_lio_opcode,
+#if defined(HAVE_POSIXAIO)
```
Comment (xiexingguo) [inline src/os/bluestore/KernelDevice.cc]: Does not build:

```
/home/jenkins-build/build/workspace/ceph-pull-requests/src/os/bluestore/KernelDevice.cc:527:25: error: 'struct aio_t' has no member named 'aio'; did you mean 'aio_t'?
                 aio[i]->aio.iocb.aio_lio_opcode,
                         ^~~
                         aio_t
src/os/CMakeFiles/os.dir/build.make:782: recipe for target 'src/os/CMakeFiles/os.dir/bluestore/KernelDevice.cc.o' failed
make[3]: *** [src/os/CMakeFiles/os.dir/bluestore/KernelDevice.cc.o] Error 1
```

**Comment by tchaikov** [comment]
retest this please

File: src/os/bluestore/KernelDevice.cc (hunk 1 above)
Comment (wjwithagen) [reply src/os/bluestore/KernelDevice.cc]: @xiexingguo 
I noticed, and fixed.
But now the error is in some seastar test...

Comment (tchaikov) [reply src/os/bluestore/KernelDevice.cc]: @wjwithagen it's fixed. sorry for the inconvenience.

**Comment by wjwithagen** [comment]
This would also need a backport to nautilus.

**Comment by wjwithagen** [comment]
@badone 
Hi Brad,
You closed the backport  tracker for this PR as duplicate.
Whilest I was led to believe that the tracker was the way to go.
So how do I get this one and #26799 backported?

**Comment by badone** [comment]
> @badone
> Hi Brad,
> You closed the backport tracker for this PR as duplicate.
> Whilest I was led to believe that the tracker was the way to go.
> So how do I get this one and #26799 backported?

@wjwithagen Oh, really? Sorry. That's not how backport PRs are done though. Take a look at some existing backport trackers as well as the backport related scripts in src/script/ for details. Let me know if you need assistance!

**Comment by wjwithagen** [comment]
> 
> 
> > @badone
> > Hi Brad,
> > You closed the backport tracker for this PR as duplicate.
> > Whilest I was led to believe that the tracker was the way to go.
> > So how do I get this one and #26799 backported?
> 
> @wjwithagen Oh, really? Sorry. That's not how backport PRs are done though. Take a look at some existing backport trackers as well as the backport related scripts in src/script/ for details. Let me know if you need assistance!

@badone 
I got the following instructions some time ago:
    https://github.com/ceph/ceph/pull/26277#issuecomment-460671923
But I'll look into your suggestions as well. 
`./src/script/backport-create-issue` is complaining it misses `redminelib`, but I tried installing a few packages that I could find with `pip search` but it seems none of them satisfy the missing lib.
Suggestions as to what package I need to install for this?

**Comment by tchaikov** [comment]
@wjwithagen have you tried 
```
pip3 install python-redmine
```
?

**Comment by wjwithagen** [comment]
> 
> 
> @wjwithagen have you tried
> 
> ```
> pip3 install python-redmine
> ```
@tchaikov
That was the first I tried..... 
IF that is the one I need, I'll go and figure out why it does not help. Could be py2 <> py3.

Right it is all about the versioning mess. It hasn't made life any easier.

**Comment by badone** [comment]
> @badone
> I got the following instructions some time ago:
> [#26277 (comment)](https://github.com/ceph/ceph/pull/26277#issuecomment-460671923)

Right, following those instructions will create actual backport trackers (you'll see what I mean once you create them).

**Pull request merged**